vpr/src/route/rr_graph.cpp

```diff
@@ -380,10 +380,6 @@ void create_rr_graph(const t_graph_type graph_type,
 
     print_rr_graph_stats();
 
-    if (router_lookahead_type == e_router_lookahead::MAP) {
-        compute_router_lookahead(segment_inf.size());
-    }
```

Comment:
The main motivation of computing the lookahead here was to re-compute it only when the RR graph is changed, and not each time the router is invoked. Does this PR change that behaviour?

Reply:
It did, I've added this behavior back in the form of an explicit lookahead cache.  I think this approach scales to accommodating more lookaheads (and keeping them isolated).